src/object.cr

```diff
@@ -355,11 +359,19 @@ class Object
           {{var_prefix}}\{{name.var.id}} : \{{name.type}}?
 
           def {{method_prefix}}\{{name.var.id}}
-            {{var_prefix}}\{{name.var.id}} ||= \{{yield}}
+            if {{var_prefix}}\{{name.var.id}}.nil?
+              {{var_prefix}}\{{name.var.id}} = \{{yield}}
+            else
+              {{var_prefix}}\{{name.var.id}}.not_nil!
```

Comment:
This (and all) `not_nil!` should be avoided because it will check again for nil, and also generate code to raise in case it's nil. It's better to assign the ivar to a local variable first.

Reply:
Thanks for the tip! I forgot that I'm dealing here with with c/ivars and out the frustration tucked `not_nil!` all over the place ;)

In the updated code I named the local variable after the c/ivar name + `_` prefix to avoid clashing with language keywords, yet I'm thinking whether would be cleaner to use some generic name like `value` instead, what do you think?

Reply:
@Sija what about a fresh variable `%value` ?